spec/features/recipes/new_recipe_page_spec.rb

```diff
@@ -0,0 +1,66 @@
+feature 'user creates new recipe' do
+  before do
+    FactoryBot.create(:recipe, :scrambled_eggs)
+    visit('/recipes/new')
+  end
+
+  context 'when valid params are submitted' do
+    before do
+      fill_in('recipe_name', with: 'Example name')
+      fill_in('recipe_instructions', with: 'Example instructions')
+      page.select 'egg', from: 'recipe_recipe_ingredients_attributes_0_ingredient_id'
+      fill_in('recipe_recipe_ingredients_attributes_0_unit_amount', with: '120')
+      page.select 'butter', from: 'recipe_recipe_ingredients_attributes_1_ingredient_id'
+      fill_in('recipe_recipe_ingredients_attributes_1_unit_amount', with: '10')
+      click_button('Create Recipe')
+    end
+
+    it 'creates new recipe' do
+      expect(page).to have_content('Successfully created a recipe')
+      expect(page).to have_content('egg')
+      expect(page).to have_content('butter')
+    end
+  end
+
+  context 'when empty params are submitted' do
+    before do
+      fill_in('recipe_name', with: '')
+      fill_in('recipe_instructions', with: '')
+      click_button('Create Recipe')
+    end
+
+    it 'displays presence errors' do
+      current_path.match(new_recipe_path)
+      expect(page).to have_content("can't be blank")
+    end
+  end
+
+  context 'when too short params are submitted' do
+    before do
+      fill_in('recipe_name', with: 'e')
+      fill_in('recipe_instructions', with: 'e')
+      click_button('Create Recipe')
+    end
+
+    it 'displays length error' do
+      current_path.match(new_recipe_path)
+      expect(page).to have_content('is too short')
+    end
+  end
+
+  context 'when ingredients params are invalid' do
```

Comment:
I think that we are going too much into details. Integration tests are not supposed to be testing stuff like every single validation on model. Integration tests are expensive (in terms of execution time). It is better to test just one case and test details by writing unit tests. I'm fine if you leave it like this, just telling you that it would be nice if you could start writing unit tests.